src/compression/zip.rs

```diff
@@ -8,23 +8,24 @@
 use super::*;
 use super::optimize_bytes::*;
 
-use std::io::{self, Read};
-use libflate::zlib::{Encoder, Decoder};
+use std::io;
 use crate::error::Result;
+use deflate::write::ZlibEncoder;
+use inflate::{InflateWriter};
+use io::Write;
 
 // scanline decompression routine, see https://github.com/openexr/openexr/blob/master/OpenEXR/IlmImf/ImfScanLineInputFile.cpp
 // 1. Uncompress the data, if necessary (If the line is uncompressed, it's in XDR format, regardless of the compressor's output format.)
 // 3. Convert one scan line's worth of pixel data back from the machine-independent representation
 // 4. Fill the frame buffer with pixel data, respective to sampling and whatnot
 
 
-pub fn decompress_bytes(data: Bytes<'_>, expected_byte_size: usize) -> Result<ByteVec> {
-    let mut decompressed = Vec::with_capacity(expected_byte_size);
+pub fn decompress_bytes(data: Bytes<'_>, _expected_byte_size: usize) -> Result<ByteVec> {
+    //let mut decompressed = Vec::with_capacity(expected_byte_size);
 
-    {
-        let mut decompressor = Decoder::new(data)?;
-        decompressor.read_to_end(&mut decompressed)?;
-    };
+    let mut decompressor = InflateWriter::new(Vec::new());
```

Comment:
I was trying to avoid allocating too much, of course doing lots allocations afterwards isn't also good. On my tests with multilayer files with zip compression, I didn't notice any increase in time, and it's faster than the "official" OpenEXR version. Again, I'm more than happy to change that so it allocates the memory in advance.

Reply:
It's probably not necessary, I believe you. Thanks for changing it anyways.